### PR TITLE
Remove unsupported rubies

### PR DESCRIPTION
**Pull request by kolorahl**
Ruby 2.3 was EOL'd on March 31, 2019. As of now, only Rubies 2.4 and above are officially supported, so those are the rubies we will focus our efforts to support.

Also: Removed the patch versions to have Travis install whatever the most recent patch version is for a given major/minor version. For example, although Ruby 2.5.5 exists, Travis CI doesn't support it presently, so it's better to just specify 2.5 and let the built-in Travis alias determine the most recent patch version it supports.

### CODE REVIEW

**Comment by kolorahl** [comment]
For some reason Ruby 2.5 is just not working. Doesn't seem to have bundler (or the right version of it) installed, although I'm dubious about including a manual "install bundler" step in the Travis config. Seems janky.

**Comment by kolorahl** [comment]
Interestingly enough it seems like this is not an uncommon issue. See [travis-ci/issues/8969](https://github.com/travis-ci/travis-ci/issues/8969) for details.

TL;DR

Some people were/are having this problem with Ruby 2.4.2 as well as 2.5.x. The solution has, to date, seemingly been to either accept Ruby 2.5 as a possible failing Ruby target or to simply remove 2.5 completely from the config. I'm opting for the latter since this issue is over a year old and I'm not sure it's getting officially fixed by travis-ci at this point.

**Comment by kolorahl** [comment]
Ran into some rubocop warnings that were easily fixed, so I fixed them. Especially since they were actively failing my build. Exlucding `lib/amplitude-api.rb` from the file naming cop as well as auto-correcting the "frozen string literal" warnings.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 language: ruby
 cache: bundler
 rvm:
-  - 2.6.1
-  - 2.4.6
-  - 2.3.8
-  - 2.2.10
+  - 2.6
+  - 2.4
```
Comment (kolorahl) [inline .travis.yml]: For some reason, version 2.5 doesn't seem to work, at all. Not sure the correct way to resolve this but because 2.5 wasn't being targeted previously, I'm not going to consider it a blocker to suddenly add support for it. The primary goal is just to remove unsupported rubies, not add additional support.
